File: solutions/0205. Isomorphic Strings/0205.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  bool isIsomorphic(string_view s, string_view t) {
    vector<int> mapS(128);
    vector<int> mapT(128);

    for (int i = 0; i < s.length(); ++i) {
      if (mapS[s[i]] != mapT[t[i]])
        return false;
      mapS[s[i]] = i + 1;
      mapT[t[i]] = i + 1;    
    }

    return true;
  }
};

File: solutions/0205. Isomorphic Strings/0205.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def isIsomorphic(self, s: str, t: str) -> bool:
    mapS = [0] * 128
    mapT = [0] * 128

    for i in range(len(s)):
      if mapS[ord(s[i])] != mapT[ord(t[i])]:
        return False
      mapS[ord(s[i])] = i + 1
      mapT[ord(t[i])] = i + 1

    return True